Documentation/gettingstarted/tls-visibility.rst

```diff
@@ -0,0 +1,308 @@
+
+.. parsed-literal::
+
+    $ kubectl create -f \ |SCM_WEB|\/examples/kubernetes-dns/dns-sw-app.yaml
+    $ kubectl get po
+    NAME                             READY   STATUS    RESTARTS   AGE
+    pod/mediabot                     1/1     Running   0          14s
+
+
+A Brief Overview of the TLS Certificate Model
+=============================================
+
+TLS is a protocol that "wraps" other protocols like HTTP and ensures that communication between client and
+server has confidentiality (no one can read the data except the intended recipient), integrity (recipient
+can confirm that the data has not been modified in transit), and authentication (sender can confirm that
+it is talking with the intended destination, not an imposter).  We will provide a highly simplified overview
+of TLS in this document, but for full details, please see XXX .
+
+From an authentication perspective, the TLS model relies on a "Certificate Authority" (CA) which is an
+entity that is trusted to create proof that a given network service (e.g., www.cilium.io)
+is who they say they are.   The goal is to prevents a malicious party in the network between the client
+and the server from intercepting the traffic and pretending to be the destination server.
+
+In the case of "friendly interception" for network security monitoring, Cilium uses a model similar to
+traditional firewalls with TLS inspection capabilities:  the network security team creates their own "internal
+certificate authority"
+that can be used to create alternative certificates for external destinations.  This model requires each
+client workload to also trust this new certificate, otherwise the client's TLS library will reject
+the connection as invalid.  In this model, the network firewall uses the certificate signed by the internal
+CA to act like the destination service and terminate the TLS connection.  This allows the firewall to
+inspect and even modify the application layer data, and then initiate another TLS connect to the actual
+destination service.
+
+The CA model within TLS is based on cryptographic keys and certificates.  Realizing the above model
+requires four primary steps:
+
+1) Create an internal certificate authority by generating a CA private key and CA certificate.
+
+2) For any destination where TLS inspection is desired (e.g., swapi.co in the example below), generate a
+private key and certificate signing request with a common name that matches the destination DNS name.
+
+3) Use the CA private key to create a signed certificate.
+
+4) Ensure that all clients where TLS inspection is have the CA certificate installed so that they will
+trust all certificates signed by that CA.
+
+5) Given that Cilium will be terminating the initial TLS connection from the
+client and creating a new TLS connection to the destination, Cilium must be told the set of CAs that it
+should trust when validating the new TLS connection to the destination service.
```

Comment:
Having a graphical example that shows where each step is would be really excellent.

Reply:
Agreed, I'll try to whip something together.